src/include/parser/expression/abstract_expression.h

```diff
@@ -102,6 +102,19 @@ class AbstractExpression {
     return children_[index];
   }
 
+  /**
+   * Exact equality check, including value equality
+   * @param other
+   * @return true if the two expressions are exactly equal
+   */
+  bool ExactlyEquals(const AbstractExpression &other) const {
```

Comment:
It's unclear what `==` is vs. this `ExactlyEquals`. It probably is better to write more detailed comments explaining the semantics. Better still, maybe it is better to not override `==` and instead just write two methods. (e.g. `ShallowEquals` and `DeepEquals` with documentation explaining the difference)

Reply:
This method is not needed for plan nodes. It was required for parts of the optimizer. I think I am going to take it out for now and investigate whether we need it later.